src/main/scala/fs2/interop/cats/cats.scala

```diff
@@ -24,6 +27,15 @@ package object cats extends Instances {
     def runFoldMap[B](f: A => B)(implicit M: Monoid[B]): Free[F, B] =
       self.runFold(M.empty)((b, a) => M.combine(b, f(a)))
 
+    def runGroupByFoldMap[K, B: Monoid](f: A => K)(g: A => B): Free[F, Map[K, B]] =
+      runFoldMap(a => Map(f(a) -> g(a)))
+
+    def runGroupByFoldMonoid[K](f: A => K)(implicit M: Monoid[A]): Free[F, Map[K, A]] =
+      runFoldMap(a => Map(f(a) -> a))
+
+    def runGroupBy[X[_], K](f: A => K)(implicit A: Applicative[X], M: Monoid[X[A]]): Free[F, Map[K, X[A]]] =
```

Comment:
Note that inference of `F` will fail often here, requiring folks to specify the type params manually. I think it is likely better to not include this method but I don't feel very strongly about that. What do you think?

```
scala> Stream(1, 2, 3).covary[Task].runGroupBy(a => a % 2)
<console>:25: error: ambiguous implicit values:
 both value Id in package cats of type => cats.Bimonad[cats.Id] with cats.Traverse[cats.Id]
 and value function0Instance in trait Function0Instances of type => cats.Bimonad[Function0]
 match expected type cats.Applicative[X]
       Stream(1, 2, 3).covary[Task].runGroupBy(a => a % 2)
                                              ^

scala> Stream(1, 2, 3).covary[Task].runGroupBy[Id, Int](a => a % 2).run.unsafeRun
res11: Map[Int,cats.Id[Int]] = Map(1 -> 4, 0 -> 2)
```